lib/aws/active_job/sqs/job_runner.rb

```diff
@@ -7,20 +7,68 @@ module SQS
       class JobRunner
         attr_reader :id, :class_name
 
+        def self.queue_event_handlers
+          @@queue_handlers ||= {}.tap do |handlers|
+            Aws::ActiveJob::SQS.config.queues.values.each do |queue_config|
+              next unless queue_config[:event_message_class].present?
+
+              handlers[queue_config[:url]] = queue_config[:event_message_class]
+            end
+          end
+        end
+
         def initialize(message)
-          @job_data = ActiveSupport::JSON.load(message.data.body)
+          @job_data   = prepare_job_data(message)
           @class_name = @job_data['job_class'].constantize
-          @id = @job_data['job_id']
+          @id         = @job_data['job_id']
         end
 
         def run
-          ::ActiveJob::Base.execute @job_data
+          ::ActiveJob::Base.execute(@job_data)
         end
 
         def exception_executions?
           @job_data['exception_executions'] &&
             !@job_data['exception_executions'].empty?
         end
+
+        private
+
+        def prepare_job_data(message)
+          return ActiveSupport::JSON.load(message.data.body) if active_job_message?(message)
+
+          format_event_data(message)
+        end
+
+        def format_event_data(message)
+          {
+            'job_class' => job_class_from_config(message.queue_url),
+            'job_id' => message.message_id,
+            'arguments' => [
+              message.data.as_json.merge(
+                'receipt_handle' => message.receipt_handle,
+                'queue_url' => message.queue_url
+              )
+            ]
+          }
+        end
+
+        # Active job messages will have message_attributes key 'aws_sqs_active_job_class'
+        def active_job_message?(message)
+          !message
```

Comment:
I think you can just check `message.message_attributes.key?('aws_sqs_active_job_class')` could you not?